app/src/main/java/org/simple/clinic/summary/PatientSummaryScreen.kt

```diff
@@ -145,6 +153,7 @@ class PatientSummaryScreen(context: Context, attrs: AttributeSet) : RelativeLayo
   private fun bloodPressureSaves() = screenRouter.streamScreenResults()
       .ofType<ActivityResult>()
       .filter { it.requestCode == REQCODE_BP_SAVED && it.succeeded() }
+      .doOnNext { bloodPressureWasSaved = BloodPressureEntrySheet.wasBloodPressureSaved(it.data!!) }
       .map { PatientSummaryBloodPressureClosed(BloodPressureEntrySheet.wasBloodPressureSaved(it.data!!)) }
```

Comment:
Instead of extracting the result from the `Intent` twice, we can instead add another map:

```kotlin
private fun bloodPressureSaves() = screenRouter.streamScreenResults()
      .ofType<ActivityResult>()
      .filter { it.requestCode == REQCODE_BP_SAVED && it.succeeded() }
      .map {  BloodPressureEntrySheet.wasBloodPressureSaved(it.data!!) }
      .doOnNext { saved -> bloodPressureWasSaved = saved }
      .map { saved -> PatientSummaryBloodPressureClosed(saved) }
```